src/loan.rs

```diff
@@ -205,32 +355,72 @@ impl CollateralContract {
         let ident_pk_sat = {
             let mut hash_map = HashMap::new();
 
-            let script = &self.descriptor.explicit_script();
+            let script = &self.raw_script;
             let sighash = SigHashCache::new(&*transaction).segwitv0_sighash(
                 input_index as usize,
                 &script,
                 input_value,
                 SigHashType::All,
             );
-            let sighash = elements::secp256k1_zkp::Message::from(sighash);
+            let sighash = secp256k1_zkp::Message::from(sighash);
 
             let sig = identity_signer(sighash)
                 .await
                 .context("could not sign on behalf of lender")?;
-            hash_map.insert(self.lender_pk, (sig, SigHashType::All));
+            hash_map.insert(identity_pk, (sig, SigHashType::All));
 
             hash_map
         };
 
-        let after_sat = After(self.timelock as u32);
+        Ok((cov_sat, cov_pk_sat, ident_pk_sat))
+    }
 
-        self.descriptor.satisfy(
-            &mut transaction.input[input_index as usize],
-            (cov_sat, cov_pk_sat, ident_pk_sat, after_sat),
-        )?;
+    /// Satisfy the covenant descriptor based on the satisfier
+    /// provided, and modify the value returned by
+    /// `elements-miniscript` to account for the rest of the contract.
```

Comment:
Maybe we can mention here that this is an extreme hack because we can't express everything with miniscript (yet).

Reply:
Done here: https://github.com/comit-network/baru/pull/25/commits/a7743b538c91ff3f055c3a1f22a14502be7e275b